datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -87,6 +88,19 @@
 if TYPE_CHECKING:
     import ssl  # noqa: F401
 
+if os.environ.get("GOFIPS", None):
+    import sys
+
+    from cryptography.hazmat.backends import default_backend
+
+    path_to_embedded = sys.executable.split("embedded")[0] + "embedded"
+    # The cryptography package can enter FIPS mode if its internal OpenSSL
+    # can access the FIPS module and configuration.
+    os.environ["OPENSSL_CONF"] = path_to_embedded + "/ssl/openssl.cnf"
+    os.environ["OPENSSL_MODULES"] = path_to_embedded + "/lib/ossl-modules"
+    cryptography_backend = default_backend()
+    cryptography_backend._enable_fips()
```

Comment:
Should we throw an exception if FIPS mode was not actually enabled? Is there a _get_fips_mode()? We could also log a warning but maybe  proceeding in non-FIPS mode is worse than failing to start as it would make us non-compliant.

Reply:
`cryptography_backend._enable_fips()` errors out with a `cryptography.exceptions.InternalError` related to OpenSSL in the case where the function does not run.
We could catch that and print out a custom message alongside it.

There is a [_fips _enabled](https://github.com/pyca/cryptography/blob/6a5cb96832088e8a0f76994f76473470c0811aae/src/rust/cryptography-openssl/src/fips.rs#L13) function that we can use as a second check as well.